src/comodo/jaxsimSimulator/jaxsimSimulator.py

```diff
@@ -0,0 +1,465 @@
+
+        # Integrator (not used for visco-elastic contacts)
+        self._integrator: Optional[integrators.common.Integrator] = None
+
+        # Integrator state for the simulation (not used for visco-elastic contacts)
+        self._integrator_state: Optional[dict[str, Any]] = None
+
+        # Time step for the simulation
+        self._dt: Optional[float] = None
+
+        # Joint torques
+        self._tau: Optional[npt.ArrayLike] = None
+
+        # Link contact forces
+        self._link_contact_forces: Optional[npt.ArrayLike] = None
+
+        # Contact model to use
+        self._contact_model_type: ContactModelEnum = ContactModelEnum.RELAXED_RIGID
+
+        # Index of the left foot link
+        self._left_foot_link_idx: Optional[int] = None
+
+        # Index of the right foot link
+        self._right_foot_link_idx: Optional[int] = None
+
+        # Index of the left foot sole frame
+        self._left_footsole_frame_idx: Optional[int] = None
+
+        # Index of the right foot sole frame
+        self._right_footsole_frame_idx: Optional[int] = None
+
+        # ==== Visualization attributes ====
+        # Mode for visualization ('record', 'interactive', or None)
+        self._visualization_mode: Optional[str] = None
+
+        # Visualization object
+        self._viz: Optional[MujocoVisualizer] = None
+
+        # Recorder for video recording
+        self._recorder: Optional[MujocoVideoRecorder] = None
+
+        # Frames per second for visualization
+        self._viz_fps: int = 10
+
+        # Last rendered time in nanoseconds
+        self._last_rendered_t_ns: float = 0.0
+
+        # Optional Mujoco model helper
+        self._mj_model_helper: Optional[MujocoModelHelper] = None
+
+    # ==== Simulator interface methods ====
+
+    def load_model(
+        self,
+        robot_model: RobotModel,
+        *,
+        dt: float = 0.001,
+        xyz_rpy: npt.ArrayLike = np.zeros(6),
+        contact_model_type: ContactModelEnum = ContactModelEnum.RELAXED_RIGID,
+        s: Optional[npt.ArrayLike] = None,
+        contact_params: Optional[ContactsParams] = None,
+        left_foot_link_name: Optional[str] = "l_ankle_2",
+        right_foot_link_name: Optional[str] = "r_ankle_2",
+        left_foot_sole_frame_name: Optional[str] = "l_sole",
+        right_foot_sole_frame_name: Optional[str] = "r_sole",
+        visualization_mode: Optional[str] = None,
+    ) -> None:
+        # ==== Initialize simulator model and data ====
+
+        self._dt = dt
+        self._contact_model_type = contact_model_type
+
+        match contact_model_type:
+            case ContactModelEnum.RIGID:
+                contact_model = RigidContacts()
+            case ContactModelEnum.RELAXED_RIGID:
+                contact_model = RelaxedRigidContacts()
+            case ContactModelEnum.VISCO_ELASTIC:
+                contact_model = ViscoElasticContacts()
+            case ContactModelEnum.SOFT:
+                contact_model = SoftContacts()
```

Comment:
I suggest to use the `build()` methods to build the contact models.

Reply:
Thanks!